semgrep-core/src/engine/Match_rules.ml

```diff
@@ -129,6 +129,8 @@ type env = {
   pattern_matches : id_to_match_results;
   (* unused for now, but could be passed down for Range.content_at_range *)
   file : Common.filename;
+  rule : R.rule;
```

Comment:
maybe a comment explaining why we need the rule and xlang in the enviroment (I guess for metavariable-pattern)?